app/manageEvent/[id]/page.tsx

```diff
@@ -0,0 +1,31 @@
+'use client';
+
+import { useState, useEffect } from "react";
+import { Event } from "@prisma/client";
+import RouteButton from "@/components/RouteButton";
+
+export default function ManageEvent({ params }: { params: { id: string } }) {
+	const [event, setEvent] = useState<Event>();
+	
+	useEffect(() => {
+		const getEvent = async () => {
+			const id = params.id;
+			const res = await fetch(`/api/event?id=${id}`, { method: 'GET' });
+			const { event } = await res.json();
+			setEvent(event);
+		}
+		getEvent();
```

Comment:
What's this for?

Reply:
load specific event (by id) from api

Reply:
Why are you calling the function inside the useEffect?

Reply:
to call that function when the page loads, idk if there's another way

Reply:
I'm pretty sure it gets called automatically by useEffect, the array parameter are the dependencies of the useEffect and since it is empty that function should be called every time on a page load without having to call it.

Reply:
ok so it still doesn't get called automatically by useEffect, ig it just treats the function definition as just a definition, and that I still have to explicitly call it

Reply:
try moving the code outside the function

Reply:
wym, like this?

```typescript
	const getEvents = async () => {
		const res = await fetch('/api/event', { method: 'GET' });
		const { events } = await res.json();
		setEvents(events);
	}

	useEffect(() => {
		getEvents();
	}, []);
```

b/c if it's like this, it can also work, but that's just the same way I had it

if that's not what you meant then copy/edit that code snippet and send it for me to test

Reply:
I meant taking the code inside the function and putting it inside the useeffect and removing the function.

Reply:
```
	useEffect(() => {
		const res = await fetch('/api/event', { method: 'GET' });
		const { events } = await res.json();
		setEvents(events);
	}, []);

```

Reply:
```typescript
	useEffect(async () => {
	    const res = await fetch('/api/event', { method: 'GET' });
	    const { events } = await res.json();
	    setEvents(events);
	}, []);
```

if you mean like this^

Then I'd get an error for using an `async` on useEffect b/c TypeScript is not expecting a `Promise` as a return. But let's say I remove the `async`, then I won't be able to make the `await fetch` calls.

Reply:
> useEffect(() => {
> 		const res = await fetch('/api/event', { method: 'GET' });
> 		const { events } = await res.json();
> 		setEvents(events);
> 	}, []);

ye i ended up realizing what you meant, this is actually what I originally tried, so i looked up a fix and just called the func

Reply:
Make that change so we can merge this

Reply:
idk if u saw the reply, but when u gave me the option to add the hook w/ the optional param, I decided that since the `useEffect` calls to get event(s) will likely not be called again, there's not much benefit to creating the hook

but I reassure that if we get to a point where we will need to make those specific calls again throughout the project, then I'll add the hook